client/blocks/inline-help/inline-help-search-results.jsx

```diff
@@ -13,8 +13,6 @@ import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { getCurrentUserId } from 'calypso/state/current-user/selectors';
 import getAdminHelpResults from 'calypso/state/inline-help/selectors/get-admin-help-results';
 import getContextualHelpResults from 'calypso/state/inline-help/selectors/get-contextual-help-results';
-import hasCancelableUserPurchases from 'calypso/state/selectors/has-cancelable-user-purchases';
```

Comment:
It may be worth leaving a comment where `<QueryUserPurchases userId={ currentUserId } />` is to explain it's needed for the selector? I could imagine it being confusing and out-of-place to someone else who takes a look at this in the future.

Reply:
Maybe filtering the _Managing purchases_ link shouldn't even be part of the selector because it somehow hides that dependency. @jsnajdr what do you think about moving it back to component space but still untangle it from the other search result types?

Reply:
The potentially confusing thing here is that we must not forget to render the `<QueryUserPurchase />` component in order for this check to really work. If both the query and the selector are in the same module, it's easier to figure out what's going on.

In near future, I'd like to add a `user.has_purchases` field to the `/me` response, and that makes this bit of information always available, because the current user is always known.

Also, I have some doubts whether the UX improvement achieved by removing the "Manage Purchases" link is really worth the effort and the additional dependencies added to the, otherwise quite self-sufficient, `InlineHelp` component.